src/nni_manager/training_service/kubernetes/kubernetesJobInfoCollector.ts

```diff
@@ -23,21 +23,16 @@ export class KubernetesJobInfoCollector {
         this.statusesNeedToCheck = ['RUNNING', 'WAITING'];
     }
 
-    public async retrieveTrialStatus(kubernetesCRDClient: KubernetesCRDClient | undefined): Promise<void> {
+    public async retrieveTrialStatus(kubernetesCRDClient: KubernetesCRDClient | undefined): Promise<void[]> {
```

Comment:
why change this function?

Reply:
Hi, it's based on the following reasons:
* Firstly, it's an `async` function, all its sync usages if any should and actually do `await` outside as in the figure below, rather than before returning.

<img width="667" alt="Screen Shot 2020-10-27 at 12 43 04 PM" src="https://user-images.githubusercontent.com/21228317/97333769-49d4cd80-1852-11eb-842c-dc3187916665.png">

* Since the `Promise` is for all retrievals of the status, the return type should be `void[]` instead of `void`.
https://github.com/microsoft/nni/blob/95bb20be17adda7375128c306bfa58a8ac6e31b8/src/nni_manager/training_service/kubernetes/kubernetesJobInfoCollector.ts#L33-L35

* If one wants to have specialized semantics for Kubeflow retrieval, it should be written in the child method for `KubeflowJobInfoCollector.retrieveSingleTrialJobInfo()`. In the meanwhile, in `kubeflowTrainingService.ts`, it's already delaying before retrieval as shown below, so we made the changes you see in this file.

https://github.com/microsoft/nni/blob/95bb20be17adda7375128c306bfa58a8ac6e31b8/src/nni_manager/training_service/kubernetes/kubeflow/kubeflowTrainingService.ts#L58-L59